main.cpp

```diff
@@ -28,510 +28,592 @@ Thing: Car Wash
-    If you have warnings about 'unused parameter', you aren't using one of your function parameters in your implementation.
-    use the parameter in your implementation.
-    If you have warnings about 'overshadow', a local variable in the function has the same name as a class member.
-    change the name of your local variable so it is different than the class member's name.
- 
- 3) be sure to write the correct full qualified name for the nested type's member functions.
- 
- 4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
- */
+/*
+2) provide implementations for the member functions you declared in your 10 user-defined types from the previous video outside of your udt definitions.
+   If you have warnings about 'unused parameter', you aren't using one of your function parameters in your implementation.
+   use the parameter in your implementation.
+   If you have warnings about 'overshadow', a local variable in the function has the same name as a class member.
+   change the name of your local variable so it is different than the class member's name.
 
+3) be sure to write the correct full qualified name for the nested type's member functions.
+ // Full qualified type: include owning type, scope operator and name of the function (defining OUTSIDE the struct/class declaration)
 
+4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
+*/
 
-/*
-Thing 1) Dragon
-5 properties:
-    1) family origin
-    2) fire power
-    3) number of knights defeated in lifetime
-    4) birthplace
-    5) age
-3 things it can do:
-    1) capture princesses
-    2) breath fire
-    3) fight knights
- */
+//================================================================================
 
 struct Dragon
 {
-//    1) family origin
     std::string familyOrigin = "Smaug";
-//    2) fire power
     int firePower = 350;
-//    3) number of knights defeated in lifetime
     int knightsDefeated = 57;
-//    4) birthplace
     std::string birthplace = "Asgaard-upon-Thyme";
-//    5) age
     int age = 500;
 
-//    1) capture princesses
-    void capturePrincess(Dragon dragon);
-//    2) breath fire
-    void breathFire(int firePower);
-//    3) fight knights
-    void fightKnight(Dragon birthplace, int courageOfKnight);
+    bool capturePrincess(int dragonFirePower);
+    void breathFire(int firePower, std::string familyName);
+    bool fightKnight(int courageOfKnight);
 };
 
-/*
-Thing 2) High rise
-5 properties:
-    1) number of lifts
-    2) number of floors
-    3) number of rooms
-    4) hours per week cleaned
-    5) cost of property
-3 things it can do:
-    1) house residents
-    2) charge residents
-    3) transport residents between floors
- */
+    bool Dragon::capturePrincess(int dragonFirePower)
+{
+    if (dragonFirePower >=50)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
```

Comment:
this can be condensed to `return dragonFirePower >= 50;`
be sure to put space around your `>=` 
- [ ] revise

Reply:
Done.